pkg/generate/generate.go

```diff
@@ -332,11 +325,31 @@ func manageClone(log logr.Logger, kind, namespace, name string, clone map[string
 		// attempting to clone it self, this will fail -> short-ciruit it
 		return nil, Skip, nil
 	}
+
 	// check if the resource as reference in clone exists?
 	obj, err := client.GetResource(kind, newRNs, newRName)
 	if err != nil {
 		return nil, Skip, fmt.Errorf("reference clone resource %s/%s/%s not found. %v", kind, newRNs, newRName, err)
 	}
+
+	// check if resource to be generated exists
+	newResource, err := client.GetResource(kind, namespace, name)
+	if err == nil {
+		obj.SetUID(newResource.GetUID())
+		obj.SetSelfLink(newResource.GetSelfLink())
+		obj.SetCreationTimestamp(newResource.GetCreationTimestamp())
+		obj.SetManagedFields(newResource.GetManagedFields())
```

Comment:
Have you tested on 1.18- cluster? I remember `ManagedFields` is introduced in 1.18 cluster.

Reply:
yes

Reply:
How would this work on **1.18-** cluster if `ManagedFields` is not supported?

Reply:
@evalsocket Can you elaborate more on this?

Reply:
Tested it with v1.14 and v1.18 and it work smoothly.